### PR TITLE
Package and deploy the extension

### LINKED ISSUE #6: Create python package with the domain
Having the Sphinx Verilog Domain python package will be useful for easy domain integration with different Sphinx projects. Currently, the proper domain usage requires appending the path to the domain in the Sphinx configuration.

### PR DESCRIPTION
**Pull request by rw1nkler**
This PR creates the package with the `sphinx-verilog-domain` and adds the ability to deploy it to PyPI using Travis CI.

Additionally, the following changes were introduced:
  - Added make-env for environment management
  - Added basic Sphinx documentation structure
  - All the tests have been moved to Sphinx documentation

The PR requires proper token setup for PyPI

Resolves https://github.com/SymbiFlow/sphinx-verilog-domain/issues/6

### CODE REVIEW

File: sphinx-verilog-domain/verilogdomain.py [hunk 1]
```diff
@@ -662,13 +662,3 @@ def resolve_xref(self, env, fromdocname, builder, typ, target, node, contnode):
         qualified_name_without_root = VerilogQualifiedIdentifier(obj.qualified_name[1:])
         tooltip = f"{obj.objtype} {qualified_name_without_root}" if obj.objtype else f"{qualified_name_without_root}"
         return make_refnode(builder, fromdocname, obj.docname, obj.linktarget, contnode, tooltip)
-
```
Comment (mithro) [inline sphinx-verilog-domain/verilogdomain.py]: Why are you removing this?

Comment (rw1nkler) [reply sphinx-verilog-domain/verilogdomain.py]: This part has been moved to the [`__init__.py` file](https://github.com/SymbiFlow/sphinx-verilog-domain/pull/13/files#diff-bf04fac69205a9863fcfebcea57819a5R4) in the same fashion as it is done for [sphinxcontrib-verilog-diagrams](https://github.com/SymbiFlow/sphinxcontrib-verilog-diagrams/blob/master/sphinxcontrib_verilog_diagrams/__init__.py#L476) and i.e. [Breathe extension](https://github.com/michaeljones/breathe/blob/master/breathe/__init__.py#L10).

File: README.rst [hunk 2]
```diff
@@ -17,7 +15,7 @@ Enabling
 
 Add extension in ``conf.py``::
 
-    extensions = [ 'sphinxcontrib.verilogdomain' ]
+    extensions = [ 'sphinx-verilog-domain' ]
```
Comment (mithro) [inline README.rst]: Why did you change the name of the package?

File: docs/requirements.txt [hunk 3]
```diff
@@ -0,0 +1,5 @@
+git+https://github.com/SymbiFlow/sphinx_symbiflow_theme
+
+sphinx
+symbolator
+sphinxcontrib-verilog-diagrams
```
Comment (mithro) [inline docs/requirements.txt:5]: This should reference the local package, otherwise it will be downloaded from PyPi?
```suggestion
-e ../
```

**Review by mithro — APPROVED** [review]
Just one comment.

Poke me after merging and I'll add the token needed to upload to PyPi.

File: docs/requirements.txt (hunk 3 above)
Comment (rw1nkler) [reply docs/requirements.txt:5]: This is about `sphinxcontrib-verilog-diagrams` not about `sphinx-verilog-domain`.

Nevertheless, the relative dependencies are not handled properly by conda. If you add the `-e ../` to the requirements file, the main environment (created by make-env) will not be created. Conda seems to handle relative dependencies in the requirements file, as a relative to the environment.yml not to the requirements file itself:

```
Pip subprocess error:
ERROR: File "setup.py" not found. Directory cannot be installed in editable mode: /mnt/hdd1//verilog_domain_ws
```

The repository is placed in:
```
/mnt/hdd1/verilog_domain_ws/sphinx-verilog-domain
```

The `sphinx-verilog-domain` package is installed in the `environment.yml` directly.
Since the RTD build also uses conda and the documentation build should happen inside this environment, I think that the `-e ../` can be omitted in the `docs/requirements.txt`

File: README.rst (hunk 2 above)
Comment (rw1nkler) [reply README.rst]: I change the directory structure to be consistent with other Sphinx extensions like:
- sphinxcontrib-verilog-diagrams
- markdown-code-symlinks

The new package name is a result of that change.

File: docs/requirements.txt (hunk 3 above)
Comment (rw1nkler) [reply docs/requirements.txt:5]: @mithro do you agree to left those files in that way? The problem with this convention will never occur unless somebody will decide to build the documentation without using `make-env` for this repository. This is exactly the same way as it is done for `sphinxcontrib-verilog-diagrams`.

Comment (mithro) [reply docs/requirements.txt:5]: @rw1nkler - I was sure that I had solved this somewhere...

Comment (mithro) [reply docs/requirements.txt:5]: There is a bunch of discussion at https://github.com/pypa/pip/issues/328

Comment (rw1nkler) [reply docs/requirements.txt:5]: It seems that this problem is not resolved for the `requirements.txt` (It will be fixed by the pipfile):
- https://github.com/pypa/pip/issues/6112
- https://github.com/pypa/pip/issues/1795

As I said before, the problem with the current settings will happen only if somebody will build the docs outside the make-env environment, which is generally wrong. I think that we should leave the requirements file as they are now.

**Comment by rw1nkler** [comment]
@mithro It seems that the problem with the editable packages installed within nested `requirements.txt` files is still unresolved. However, it should not affect documentation building and package development unless somebody will decide not to use make-env for this repository and install packages manually.

**Comment by rw1nkler** [comment]
@mithro Can we merge this PR?
